pkg/sdn/plugin/multitenant.go

```diff
@@ -0,0 +1,126 @@
+package plugin
+
+import (
+	"fmt"
+
+	log "github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	kubetypes "k8s.io/kubernetes/pkg/kubelet/container"
+	kerrors "k8s.io/kubernetes/pkg/util/errors"
+	utilwait "k8s.io/kubernetes/pkg/util/wait"
+)
+
+func (node *OsdnNode) MultitenantStartNode() error {
+	if err := node.VnidStartNode(); err != nil {
+		return err
+	}
+	if err := node.SetupEgressNetworkPolicy(); err != nil {
+		return err
+	}
+
+	go utilwait.Forever(node.watchServices, 0)
+	return nil
+}
+
+func (node *OsdnNode) updatePodNetwork(namespace string, oldNetID, netID uint32) error {
```

Comment:
updatePodNetwork() is only used by watchNetNamespaces() in vnids_node.go
I was expecting MultitenantStartNode() to watch for both NetNamespaces and Services to patch vnid when needed. I didn't understand the split between vnids_node.go and multitenant.go